File: src/main/java/insogroup/utils/utils/FilesDownloader/interfaces/FileNameGenerator.java
package insogroup.utils.utils.FilesDownloader.interfaces;

import insogroup.utils.utils.FilesDownloader.exceptions.GenerateFileNameException;

public interface FileNameGenerator {
    /**
     * Must generate the name for file
     * @param url Src url (can be used for generating name)
     * @return Name of file
     */
    String generateName(String url) throws GenerateFileNameException;
}

File: src/main/kotlin/com/insogroup/utils/FilesDownloader/interfaces/FileNameGenerator.kt
package com.insogroup.utils.FilesDownloader.interfaces

import com.insogroup.utils.FilesDownloader.exceptions.GenerateFileNameException

interface FileNameGenerator {
    /**
     * Must generate the name for file
     * @param url Src url (can be used for generating name)
     * @return Name of file
     */
    @Throws(GenerateFileNameException::class)
    fun generateName(url: String): String
}
